evm_loader/tests/test_cli.py

```diff
@@ -0,0 +1,167 @@
+    result = json.loads(
+        neon_cli().emulate(
+            evm_loader.loader_id,
+            user_account.eth_address.hex(),
+            second_user.eth_address.hex(),
+            data=None
+        )
+    )
+    assert result['exit_status'] == 'succeed', f"The 'exit_status' field is not succeed. Result: {result}"
+    assert result['steps_executed'] == 0, f"Steps executed amount is not 0. Result: {result}"
+    assert result['used_gas'] > 0, f"Used gas is less than 0. Result: {result}"
+
+
+def test_emulate_contract_deploy(user_account, evm_loader):
+    contract_path = pytest.CONTRACTS_PATH / "hello_world.binary"
+
+    with open(contract_path, 'rb') as f:
+        contract_code = f.read()
+
+    result = json.loads(
+        neon_cli().emulate(
+            evm_loader.loader_id,
+            user_account.eth_address.hex(),
+            'deploy',
+            contract_code.hex()
+        )
+    )
+    assert result['exit_status'] == 'succeed', f"The 'exit_status' field is not succeed. Result: {result}"
+    assert result['steps_executed'] > 0, f"Steps executed amount is not 0. Result: {result}"
+    assert result['used_gas'] > 0, f"Used gas is less than 0. Result: {result}"
+
+
+def test_emulate_call_contract_function(user_account, evm_loader, operator_keypair, treasury_pool):
+    contract = deploy_contract(operator_keypair, user_account, "hello_world.binary", evm_loader, treasury_pool)
+    assert contract.eth_address
+    assert get_solana_balance(contract.solana_address) > 0
+    data = abi.function_signature_to_4byte_selector('call_hello_world()')
+    result = json.loads(
+        neon_cli().emulate(
+            evm_loader.loader_id,
+            user_account.eth_address.hex(),
+            contract.eth_address.hex(),
+            data.hex()
+        )
+    )
+    assert result['exit_status'] == 'succeed', f"The 'exit_status' field is not succeed. Result: {result}"
+    assert result['steps_executed'] > 0, f"Steps executed amount is not 0. Result: {result}"
+    assert result['used_gas'] > 0, f"Used gas is less than 0. Result: {result}"
+    assert "Hello World" in to_text(result["result"])
+
+
+def test_neon_elf_params(evm_loader):
+    result = neon_cli().call(f"--evm_loader={evm_loader.loader_id} neon-elf-params").strip()
+    some_fields = ['NEON_CHAIN_ID', 'NEON_TOKEN_MINT', 'NEON_REVISION']
+    param_dict = {}
+    for param in result.split('\n'):
+        param_dict[param.split('=')[0]] = param.split('=')[1]
+    for field in some_fields:
+        assert field in param_dict, f"The field {field} is not in result {result}"
+        assert param_dict[field] != "", f"The value for fiels {field} is empty"
+
+
+def test_help():
+    result = neon_cli().call("--help").strip()
+    assert len(result) > 1000
+
+
+def test_collect_treasury(evm_loader):
+    result = neon_cli().call(f"collect-treasury --evm_loader {evm_loader.loader_id}")
+    assert len(result) > 1000
```

Comment:
I think here we need add more logic to checks. Check addresses, transaction, balances, something that show us success executed command